samples/dnn/siamrpnpp.py

```diff
@@ -0,0 +1,419 @@
+        outputs = self.model.track(x_crop)
+        score = self._convert_score(outputs['cls'])
+        pred_bbox = self._convert_bbox(outputs['loc'], self.anchors)
+
+        def change(r):
+            return np.maximum(r, 1. / r)
+
+        def sz(w, h):
+            pad = (w + h) * 0.5
+            return np.sqrt((w + pad) * (h + pad))
+
+        # scale penalty
+        s_c = change(sz(pred_bbox[2, :], pred_bbox[3, :]) /
+                     (sz(self.size[0] * scale_z, self.size[1] * scale_z)))
+
+        # aspect ratio penalty
+        r_c = change((self.size[0] / self.size[1]) /
+                     (pred_bbox[2, :] / pred_bbox[3, :]))
+        penalty = np.exp(-(r_c * s_c - 1) * self.track_penalty_k)
+        pscore = penalty * score
+
+        # window penalty
+        pscore = pscore * (1 - self.track_window_influence) + \
+                 self.window * self.track_window_influence
+        best_idx = np.argmax(pscore)
+        bbox = pred_bbox[:, best_idx] / scale_z
+        lr = penalty[best_idx] * score[best_idx] * self.track_lr
+
+        cx = bbox[0] + self.center_pos[0]
+        cy = bbox[1] + self.center_pos[1]
+
+        # smooth bbox
+        width = self.size[0] * (1 - lr) + bbox[2] * lr
+        height = self.size[1] * (1 - lr) + bbox[3] * lr
+
+        # clip boundary
+        cx, cy, width, height = self._bbox_clip(cx, cy, width, height, img.shape[:2])
+
+        # udpate state
+        self.center_pos = np.array([cx, cy])
+        self.size = np.array([width, height])
+        bbox = [cx - width / 2, cy - height / 2, width, height]
+        best_score = score[best_idx]
+        return {'bbox': bbox, 'best_score': best_score}
+
+def get_frames(video_name):
+    """
+    Args:
+        Path to input video frame
+    Return:
+        Frame
+    """
+    if not video_name:
+        cap = cv.VideoCapture(0)
+        while True:
+            ret, frame = cap.read()
+            if ret:
+                yield frame
+            else:
+                break
+    elif video_name.endswith('avi') or \
+        video_name.endswith('mp4'):
+        cap = cv.VideoCapture(video_name)
+        while True:
+            ret, frame = cap.read()
+            if ret:
+                yield frame
+            else:
+                break
+    else:
+        images = glob(os.path.join(video_name, '*.jp*'))
+        images = sorted(images,
+                        key=lambda x: int(x.split('/')[-1].split('.')[0]))
+        for img in images:
+            frame = cv.imread(img)
+            yield frame
+
+def main():
+    """ Sample SiamRPN Tracker
+    """
+    parser = argparse.ArgumentParser(description='Use this script to run SiamRPN++ Visual Tracker',
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--input_video', type=str, required=True, help='Path to input video for tracking')
```

Comment:
`parser.add_argument('--input_video', help='Path to input video file. Skip this argument to capture frames from a camera.')`